server/plugin/plugin_test.go

```diff
@@ -132,16 +148,26 @@ func TestPluginOnActivate(t *testing.T) {
 			api.On("GetConfig").Return(testutils.GetServerConfig()).Maybe()
 			defer api.AssertExpectations(t)
 
-			helpers := &plugintest.Helpers{}
-			if test.SetupHelpers != nil {
-				helpers = test.SetupHelpers(helpers)
-				defer helpers.AssertExpectations(t)
-			}
-
-			patch, _ := mpatch.PatchMethod(kvstore.NewStore, func(plugin.API, string) (store.Store, error) {
+			patch1, _ := mpatch.PatchMethod(kvstore.NewStore, func(plugin.API, string) (store.Store, error) {
 				return &mockstore.Store{}, nil
 			})
-			defer func() { require.NoError(t, patch.Unpatch()) }()
+			defer func() { require.NoError(t, patch1.Unpatch()) }()
+
+			// Setup pluginapi client
+			mClient := pluginapi.NewClient(api, &plugintest.Driver{})
+			patch2, err := mpatch.PatchMethod(
+				pluginapi.NewClient,
+				func(plugin.API, plugin.Driver) *pluginapi.Client { return mClient },
+			)
+			require.NoError(t, err)
+			defer func() { require.NoError(t, patch2.Unpatch()) }()
+
+			if test.SetupPluginAPI != nil {
+				_, patches := test.SetupPluginAPI(mClient)
+				for _, p := range patches {
+					defer func(patch *mpatch.Patch) { require.NoError(t, patch.Unpatch()) }(p)
```

Comment:
Should this use t.Cleanup()

Reply:
Thanks and welcome back @hanzei ! Fixed as suggested.